sdk/typescript/src/types/sui-bcs.ts

```diff
@@ -235,16 +235,26 @@ export type GasData = {
   budget: number;
 };
 
+/**
+ * TransactionExpiration
+ *
+ * Indications the expiration time for a transaction.
+ */
+export type TransactionExpiration = { None: null } | { Epoch: number };
+
 /**
  * The TransactionData to be signed and sent to the RPC service.
  *
  * Field `sender` is made optional as it can be added during the signing
  * process and there's no need to define it sooner.
+ *
+ * Field `expiration` is made optional as it is defaulted to `None`.
  */
 export type TransactionData = {
   sender?: string;
   kind: TransactionKind;
   gasData: GasData;
+  expiration?: TransactionExpiration;
```

Comment:
This is fine as long as it always stays null, but if it starts to become a non-null value, requests to any fullnode that does not have the rust changes in this PR will fail since they cannot recognize this new field.

There are two options for avoiding this problem:
1. make sure all fullnodes have the Rust changes in this PR(i.e., ship this with 0.x.0 devnet release, and allow non-null values in 0.x+1.0 devnet release
2. implement the changes as in Lu's PR which will serialize data based on RPC version